db/dbScript.sql

```diff
@@ -0,0 +1,393 @@
+CREATE EXTENSION IF NOT EXISTS "uuid-ossp"; --it is needed for uuid generator (uuid_generate_v4()) to be available
+
+--Basically, we dynamically create enums if they do not exist
+BEGIN
+    IF NOT EXISTS (SELECT 1 FROM pg_type WHERE typname = 'provider_type') THEN
+        CREATE TYPE provider_type AS ENUM ('local', 'google');
+    END IF;
+	IF NOT EXISTS (SELECT 1 FROM pg_type WHERE typname = 'account_type') THEN
+        CREATE TYPE account_type AS ENUM ('user', 'gym', 'admin');
+    END IF;
+	IF NOT EXISTS (SELECT 1 FROM pg_type WHERE typname = 'own_status') THEN
+        CREATE TYPE own_status AS ENUM ('pending', 'approved', 'rejected');
+    END IF;
+	IF NOT EXISTS (SELECT 1 FROM pg_type WHERE typname = 'rec_type') THEN
+        CREATE TYPE rec_type AS ENUM ('main', 'alternative');
+    END IF;
+	IF NOT EXISTS (SELECT 1 FROM pg_type WHERE typname = 'not_type') THEN
+        CREATE TYPE not_type AS ENUM ('message', 'alert', 'reminder');
+    END IF;
+END $$;
+
+--Since we will have a lot of gyms stored (potentially) it does not make sense to store working hours for each of them as a text
+--Especially since there are significantly smaller potential patterns [(opening hours, closing hours) pairs] in comparison to the potential number of gyms stored
+CREATE TABLE if not exists Working_hours
+(
+  id uuid NOT null DEFAULT uuid_generate_v4(),--uuid is analogous to id, but uuid is pseudo-random and the length and format of uuids are the same
+  open_from TIME NOT NULL,
+  open_until TIME NOT NULL,
+  PRIMARY KEY (id),
+  UNIQUE (open_from),
```

Comment:
I suggest using a UNIQUE constraint on (open_from, open_until) together. We can have to gyms with same open hours and different close hours.

Reply:
Agreed, I overlooked this)